sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+            model_url_s3 = upload_data_to_s3(input_args.upload_tarfile_s3, model_tarfile_path,
+                                            sm_session)
+            logging.info('Uploaded the model tar file to {}'.format(model_url_s3))
+
+    # ================= create deployable model ================= #
+    image_url = input_args.image_url
+    role = input_args.role
+    sm_client = boto3.client(service_name="sagemaker", region_name=input_args.region)
+
+    container = {
+        "Image": image_url,
+        "ModelDataUrl": model_url_s3,
+        "Environment": {"SAGEMAKER_PROGRAM": entrypoint_file_name}
+    }
+
+    sm_model_name = model_name + '-' + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+    create_model_response = sm_client.create_model(ModelName=sm_model_name,
+                                                   ExecutionRoleArn=role, Containers=[container])
+    logging.info('Model ARN: {}'.format(create_model_response['ModelArn']))
+
+    # ================= create an endpoint configuration ================= #
+    sm_ep_config_name = model_name + "-EndpointConfig-" + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+
+    create_endpoint_config_response = sm_client.create_endpoint_config(
+            EndpointConfigName=sm_ep_config_name,
+            ProductionVariants=[
+                {
+                    "InstanceType": input_args.instance_type,
+                    "InitialInstanceCount": input_args.instance_count,
+                    "InitialVariantWeight": 1,
+                    "ModelName": sm_model_name,
+                    "VariantName": "AllTraffic",
+                }
+            ],
+        )
+
+    logging.info("Endpoint config Arn: {}".format(create_endpoint_config_response["EndpointConfigArn"]))
+
+    # ================= create an endpoint ================= #
+    sm_ep_name = model_name + "-Endpoint-" + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+
+    create_endpoint_response = sm_client.create_endpoint(
+        EndpointName=sm_ep_name, EndpointConfigName=sm_ep_config_name
+    )
+    logging.info("Endpoint Arn: {}".format(create_endpoint_response["EndpointArn"]))
+
+    resp = sm_client.describe_endpoint(EndpointName=sm_ep_name)
+    status = resp["EndpointStatus"]
+    logging.info("Endpoint Status: " + status)
+
+    logging.info("Waiting for {} endpoint to be in service...".format(sm_ep_name))
+    waiter = sm_client.get_waiter("endpoint_in_service")
+    
+    try:
+        waiter.wait(EndpointName=sm_ep_name,
+                    WaiterConfig={
+                        'Delay': 30,        # seconds between querying
+                        'MaxAttempts': 60   # max retries (~30 minutes here)
+                    })
+        logging.info('{} endpoint has been successfully created, and ready to be \
+                 invoked!'.format(sm_ep_name))
+    except WaiterError as e:
+        logging.error("Waiter timed out or endpoint creation failed:", e)
+
```

Comment:
Shall we return sm_ep_name? So people can get the name of the endpoint after calling run_job?

Reply:
Yes. updated.

Reply:
Where is the code?
Have you uploaded the code?